pkg/kubectl/cmd/create.go

```diff
@@ -172,3 +181,54 @@ func createAndRefresh(info *resource.Info) error {
 	info.Refresh(obj, true)
 	return nil
 }
+
+// nameFromCommandArgs is a utility function for commands that assume the first argument is a resource name
+func nameFromCommandArgs(cmd *cobra.Command, args []string) (string, error) {
+	if len(args) == 0 {
+		return "", cmdutil.UsageError(cmd, "NAME is required")
+	}
+	return args[0], nil
+}
+
+// createSubcommandInternal is intended to be used by subcommands that generator client side objects
+func createSubcommandInternal(f *cmdutil.Factory, cmd *cobra.Command, name string, out io.Writer, structuredGenerator kubectl.StructuredGenerator) error {
```

Comment:
Passing the command and then directly pulling hard-coded flags from it makes it impossible to compose on top of this command if we wanted to downstream without constantly keeping every potential flag access in sync.